acceptance-tests/build.gradle

```diff
@@ -68,44 +60,46 @@ def allTestTasks =
             jenkinsVersion.default || currentJava.isCompatibleWith(jenkinsVersion.requiredJavaVersion)
         }
         .collect { jenkinsVersion ->
-            def jenkinsWarLocation =
-                new File("${project.gradle.gradleUserHomeDir}/jenkins-cache/${jenkinsVersion.version}/jenkins.war")
+            def downloadJenkinsTask =
+                tasks.register("downloadJenkins${jenkinsVersion.label}", Download) {
+                    src jenkinsVersion.downloadUrl
+                    dest file("${project.gradle.gradleUserHomeDir}/jenkins-cache/${jenkinsVersion.version}/jenkins.war")
+                    onlyIfModified true
+                    tempAndMove true
+                }
 
             def testTask =
                 jenkinsVersion.default
                     ? tasks.named("test")
-                    : tasks.register("test_${jenkinsVersion.label}", Test)
+                    : tasks.register("test${jenkinsVersion.label}", Test)
 
             testTask.configure {
-                dependsOn copyTestDependencies
+                def jenkinsWar = downloadJenkinsTask
+                def gradlePlugin = configurations.gradlePlugin
+
+                inputs.files(jenkinsWar).withPropertyName("jenkinsWar")
+                inputs.files(gradlePlugin).withPropertyName("gradlePlugin").withNormalizer(ClasspathNormalizer)
```

Comment:
❌ You need should at least select a path sensitivity if you don't use `ClasspathNormalizer`.

I think that the current declaration is turned around as well: the war file could use a `ClasspathNormalizer`, since the `.war` extension is recognized by Gradle. While the normalizer doesn't have more effect than `PathSensitivity.NONE` on `gradlePlugin`, since `.hpi` is not recognized by Gradle.

I'd add `ClasspathNormalizer` for both inputs and add a comment to `gradlePlugin` that the zip normalization doesn't work there, yet.

Reply:
Thank you for the suggestion. Applied.